src/lib/mongoBackend/MongoCommonSubscription.cpp

```diff
@@ -0,0 +1,358 @@
+
+    b->append(CSUB_HEADERS, headersObj);
+    LM_T(LmtMongo, ("Subscription headers: %s", headersObj.toString().c_str()));
+  }
+
+  if (httpInfo.qs.size() > 0)
+  {
+    BSONObjBuilder qsBuilder;
+    for (std::map<std::string, std::string>::const_iterator it = httpInfo.qs.begin(); it != httpInfo.qs.end(); ++it)
+    {
+      qsBuilder.append(it->first, it->second);
+    }
+    BSONObj qsObj = qsBuilder.obj();
+
+    b->append(CSUB_QS, qsObj);
+    LM_T(LmtMongo, ("Subscription qs: %s", qsObj.toString().c_str()));
+  }
+
+  if (httpInfo.payload != "")
+  {
+    b->append(CSUB_PAYLOAD, httpInfo.payload);
+    LM_T(LmtMongo, ("Subscription payload: %s", httpInfo.payload.c_str()));
+  }
+}
+
+/* ****************************************************************************
+*
+* setHttpInfo -
+*
+*/
+void setHttpInfo(const Subscription& sub, BSONObjBuilder* b)
+{
+  b->append(CSUB_REFERENCE, sub.notification.httpInfo.url);
+  b->append(CSUB_EXTENDED,  sub.notification.httpInfo.extended);
+
+  LM_T(LmtMongo, ("Subscription reference: %s", sub.notification.httpInfo.url.c_str()));
+  LM_T(LmtMongo, ("Subscription extended: %s", sub.notification.httpInfo.extended? "true" : "false"));
+
+  if (sub.notification.httpInfo.extended)
+  {
+    setExtendedHttpInfo(sub.notification.httpInfo, b);
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* setThrottling -
+*
+*/
+void setThrottling(const Subscription& sub, BSONObjBuilder* b)
+{
+  b->append(CSUB_THROTTLING, sub.throttling);
+  LM_T(LmtMongo, ("Subscription throttling: %lu", sub.throttling));
+}
+
+
+
+/* ****************************************************************************
+*
+* setServicePath -
+*
+*/
+void setServicePath(const std::string servicePath, BSONObjBuilder* b)
+{
+  b->append(CSUB_SERVICE_PATH, servicePath);
+  LM_T(LmtMongo, ("Subscription servicePath: %s", servicePath.c_str()));
+}
+
+
+
+/* ****************************************************************************
+*
+* setDescription -
+*
+*/
+void setDescription(const Subscription& sub, BSONObjBuilder* b)
+{
+  if (sub.description != "")
+  {
+    b->append(CSUB_DESCRIPTION, sub.description);
+    LM_T(LmtMongo, ("Subscription description: %s", sub.description.c_str()));
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* setStatus -
+*
+*/
+void setStatus(const Subscription& sub, BSONObjBuilder* b)
+{
+  std::string status = sub.status == ""?  STATUS_ACTIVE : sub.status;
+  b->append(CSUB_STATUS, status);
+  LM_T(LmtMongo, ("Subscription status: %s", status.c_str()));
+}
+
+
+
+/* ****************************************************************************
+*
+* setEntities -
+*
+*/
+void setEntities(const Subscription& sub, BSONObjBuilder* b)
+{
+  BSONArrayBuilder entities;
+  for (unsigned int ix = 0; ix < sub.subject.entities.size(); ++ix)
+  {
+    EntID en = sub.subject.entities[ix];
```

Comment:
This needs a more in-deep discussion, involving how Subscription is defined. Thus, I'd leave as it is now in the PR. If in the future Subscription changes, this would automatically change ("automatically" in the sense that changes in Subscription class would require changes in this part in order to being able to build).